### Year 1 Topology - Limit Checking - Overloading, Voltage limits

In [2]:
import pandas as pd

##### Loading

##### Percentage Loading in Branches 

Calculation related can be found in PSSE PAGVI Manual Chapter 6 Power system network simulations - Section 6.6.7 Activity Rate 

%I = MVA(actual)/(MVA(rated)*Vpu)

Vpu -> Bus Voltage at from end 

In [3]:
list_all = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
list_gen_hydro = [400,500,600] #dry, average, wet hydological scenarios
for gen in list_gens:
    for gen_hy in list_gen_hydro:
        for lsc in list_lsc:
            file_csv_out = 'savnw_sol_' + str(gen) +'_hy_' +str(gen_hy) +'_' + str(lsc)+'_l.csv'
            data_load = pd.read_csv(file_csv_out,header=1, skiprows=0)
            # Extracted data contains branch loading from the metered and non-metered ends. The data is filtered to obtain the branch loading from the metered end.
            data_fil = data_load[data_load['From Bus']!=data_load['Non Metered Bus']].reset_index(drop=True)
            data_fil['From Bus V'] = data_fil['From Bus Ext'].str.extract(r'(\d+\.\d+)') 
            data_fil['To Bus V'] = data_fil['To Bus Ext'].str.extract(r'(\d+\.\d+)')
            data_fil['Scenario']= 'Solar = ' + str(gen) + ' MW, ' +'Hydro = ' +str(gen_hy) +' MW, ' + lsc.upper()
            list_all.append(data_fil)
data_all = pd.concat(list_all).reset_index(drop=True)
data_all = data_all[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID',
       'Non Metered Bus', ' Branch I', 'Branch Flow MW', 'Branch Flow MVAR',
       'Branch MVA', 'MW Loss', 'MVAR Loss', 'Rate', '%I', '%MVA',
       'From Bus V', 'To Bus V']]

##### Checking transformer loading greater than 80% 

In [4]:
transformer_loading = data_all[data_all['From Bus V']!=data_all['To Bus V']].reset_index(drop=True)
output = transformer_loading[transformer_loading['%MVA'] > 80.0].reset_index(drop=True)
group_out = output.groupby('Scenario')
list_output_scenario = list(output['Scenario'].unique())
list_output_scenario
for list_out in list_output_scenario:
    display(group_out.get_group(list_out))

Scenario  From Bus        From Bus Ext  To Bus  \
0  Solar = 0 MW, Hydro = 400 MW, LLS       202  EAST500     500.00     203   
1  Solar = 0 MW, Hydro = 400 MW, LLS       204  SUB500      500.00     205   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
0  EAST230     230.00       1              203        733          613.40   
1  SUB230      230.00       1              205        756          638.46   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
0            219.17      651.38     1.62      65.63  800.0  79.44  81.42   
1            140.29      653.69     1.29      64.31  800.0  81.85  81.71   

  From Bus V To Bus V  
0     500.00   230.00  
1     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
2  Solar = 0 MW, Hydro = 400 MW, RLS       202  EAST500     500.00     203   
3  Solar = 0 MW, Hydro = 400 MW, RLS       204  SUB500      500.00     205   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
2  EAST230     230.00       1              203        767          632.20   
3  SUB230      230.00       1              205        785          654.33   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
2            238.77      675.79     1.77      71.75  800.0  83.06  84.47   
3            160.11      673.63     1.39      69.46  800.0  85.06  84.20   

  From Bus V To Bus V  
2     500.00   230.00  
3     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
4  Solar = 0 MW, Hydro = 400 MW, HLS       202  EAST500     500.00     203   
5  Solar = 0 MW, Hydro = 400 MW, HLS       204  SUB500      500.00     205   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
4  EAST230     230.00       1              203        814          646.79   
5  SUB230      230.00       1              205        831          666.32   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
4            284.91      706.76     1.99      80.91  800.0  88.21  88.35   
5            211.54      699.09     1.56      77.86  800.0  90.06  87.39   

  From Bus V To Bus V  
4     500.00   230.00  
5     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
6  Solar = 0 MW, Hydro = 500 MW, LLS       202  EAST500     500.00     203   
7  Solar = 0 MW, Hydro = 500 MW, LLS       204  SUB500      500.00     205   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
6  EAST230     230.00       1              203        796          670.74   
7  SUB230      230.00       1              205        844          717.23   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
6            216.08      704.68     1.90      77.28  800.0  86.20  88.09   
7            125.52      728.13     1.61      80.31  800.0  91.46  91.02   

  From Bus V To Bus V  
6     500.00   230.00  
7     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
8  Solar = 0 MW, Hydro = 500 MW, RLS       202  EAST500     500.00     203   
9  Solar = 0 MW, Hydro = 500 MW, RLS       204  SUB500      500.00     205   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
8  EAST230     230.00       1              203        841          686.76   
9  SUB230      230.00       1              205        887          730.71   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
8            258.66      733.86     2.12      86.27  800.0  91.08  91.73   
9            172.39      750.77     1.77      88.55  800.0  96.04  93.85   

  From Bus V To Bus V  
8     500.00   230.00  
9     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
10  Solar = 0 MW, Hydro = 500 MW, HLS       202  EAST500     500.00     203   
11  Solar = 0 MW, Hydro = 500 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
10  EAST230     230.00       1              203        891          701.25   
11  SUB230      230.00       1              205        936          742.61   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I   %MVA  \
10            305.88      765.06     2.38      96.87  800.0   96.51  95.63   
11            224.20      775.72     1.97      98.63  800.0  101.36  96.96   

   From Bus V To Bus V  
10     500.00   230.00  
11     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
12  Solar = 0 MW, Hydro = 600 MW, LLS       202  EAST500     500.00     203   
13  Solar = 0 MW, Hydro = 600 MW, LLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
12  EAST230     230.00       1              203        866          723.66   
13  SUB230      230.00       1              205        942          791.83   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I    %MVA  \
12            233.02      760.25     2.25      91.56  800.0   93.83   95.03   
13            133.83      803.06     2.00      99.99  800.0  102.06  100.38   

   From Bus V To Bus V  
12     500.00   230.00  
13     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
14  Solar = 0 MW, Hydro = 600 MW, RLS       202  EAST500     500.00     203   
15  Solar = 0 MW, Hydro = 600 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
14  EAST230     230.00       1              203        923          740.92   
15  SUB230      230.00       1              205        997          806.22   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I    %MVA  \
14            287.86      794.88     2.56     103.93  800.0   99.97   99.36   
15            193.76      829.18     2.24     111.94  800.0  107.98  103.65   

   From Bus V To Bus V  
14     500.00   230.00  
15     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
16  Solar = 0 MW, Hydro = 600 MW, HLS       202  EAST500     500.00     203   
17  Solar = 0 MW, Hydro = 600 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
16  EAST230     230.00       1              203        978          755.10   
17  SUB230      230.00       1              205       1052          817.78   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I   %MVA  \
16            337.66      827.16     2.87     116.67  800.0  105.92  103.4   
17            247.32      854.36     2.49     124.57  800.0  113.91  106.8   

   From Bus V To Bus V  
16     500.00   230.00  
17     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
18  Solar = 50 MW, Hydro = 400 MW, LLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
18  SUB230      230.00       1              205        740           624.3   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
18            142.82      640.43     1.23      61.66  800.0  80.14  80.05   

   From Bus V To Bus V  
18     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
19  Solar = 50 MW, Hydro = 400 MW, RLS       202  EAST500     500.00     203   
20  Solar = 50 MW, Hydro = 400 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
19  EAST230     230.00       1              203        748          619.25   
20  SUB230      230.00       1              205        765          640.35   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
19            232.45      661.44     1.68      68.32  800.0  81.05  82.68   
20            153.73      658.55     1.32      65.90  800.0  82.85  82.32   

   From Bus V To Bus V  
19     500.00   230.00  
20     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
21  Solar = 50 MW, Hydro = 400 MW, HLS       202  EAST500     500.00     203   
22  Solar = 50 MW, Hydro = 400 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
21  EAST230     230.00       1              203        795          633.89   
22  SUB230      230.00       1              205        809          652.35   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
21            277.69      692.04     1.90      77.04  800.0  86.07  86.51   
22            204.28      683.58     1.48      73.81  800.0  87.68  85.45   

   From Bus V To Bus V  
21     500.00   230.00  
22     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
23  Solar = 50 MW, Hydro = 500 MW, LLS       202  EAST500     500.00     203   
24  Solar = 50 MW, Hydro = 500 MW, LLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
23  EAST230     230.00       1              203        779          655.52   
24  SUB230      230.00       1              205        826          701.41   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
23            217.24      690.58     1.82      74.07  800.0  84.39  86.32   
24            128.58      713.09     1.54      76.92  800.0  89.51  89.14   

   From Bus V To Bus V  
23     500.00   230.00  
24     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
25  Solar = 50 MW, Hydro = 500 MW, RLS       202  EAST500     500.00     203   
26  Solar = 50 MW, Hydro = 500 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
25  EAST230     230.00       1              203        821          673.85   
26  SUB230      230.00       1              205        865          716.78   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
25            251.35      719.20     2.03      82.29  800.0  88.95  89.90   
26            165.19      735.57     1.69      84.28  800.0  93.70  91.95   

   From Bus V To Bus V  
25     500.00   230.00  
26     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
27  Solar = 50 MW, Hydro = 500 MW, HLS       202  EAST500     500.00     203   
28  Solar = 50 MW, Hydro = 500 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
27  EAST230     230.00       1              203        870          688.44   
28  SUB230      230.00       1              205        912          728.71   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
27            297.49      749.97     2.27      92.33  800.0  94.22  93.75   
28            215.96      760.03     1.87      93.73  800.0  98.81  95.00   

   From Bus V To Bus V  
27     500.00   230.00  
28     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
29  Solar = 50 MW, Hydro = 600 MW, LLS       202  EAST500     500.00     203   
30  Solar = 50 MW, Hydro = 600 MW, LLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
29  EAST230     230.00       1              203        846          710.60   
30  SUB230      230.00       1              205        921          777.83   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
29            225.49      745.52     2.15      87.43  800.0  91.69  93.19   
30            126.58      788.06     1.91      95.46  800.0  99.72  98.51   

   From Bus V To Bus V  
29     500.00   230.00  
30     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
31  Solar = 50 MW, Hydro = 600 MW, RLS       202  EAST500     500.00     203   
32  Solar = 50 MW, Hydro = 600 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
31  EAST230     230.00       1              203        902          728.12   
32  SUB230      230.00       1              205        973          792.39   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I    %MVA  \
31            279.21      779.82     2.44      99.21  800.0   97.67   97.48   
32            185.43      813.79     2.13     106.72  800.0  105.44  101.72   

   From Bus V To Bus V  
31     500.00   230.00  
32     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
33  Solar = 50 MW, Hydro = 600 MW, HLS       202  EAST500     500.00     203   
34  Solar = 50 MW, Hydro = 600 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
33  EAST230     230.00       1              203        955          742.43   
34  SUB230      230.00       1              205       1026          803.98   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I    %MVA  \
33            327.54      811.47     2.74     111.19  800.0  103.40  101.43   
34            237.61      838.36     2.37     118.47  800.0  111.09  104.79   

   From Bus V To Bus V  
33     500.00   230.00  
34     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
35  Solar = 100 MW, Hydro = 400 MW, RLS       202  EAST500     500.00     203   
36  Solar = 100 MW, Hydro = 400 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
35  EAST230     230.00       1              203        731          606.22   
36  SUB230      230.00       1              205        746          626.36   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
35            228.22      647.76     1.61      65.22  800.0  79.19  80.97   
36            149.75      644.01     1.25      62.67  800.0  80.80  80.50   

   From Bus V To Bus V  
35     500.00   230.00  
36     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
37  Solar = 100 MW, Hydro = 400 MW, HLS       202  EAST500     500.00     203   
38  Solar = 100 MW, Hydro = 400 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
37  EAST230     230.00       1              203        777          620.92   
38  SUB230      230.00       1              205        790          638.37   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
37            272.87      678.23     1.81      73.60  800.0  84.13  84.78   
38            199.74      668.89     1.40      70.22  800.0  85.52  83.61   

   From Bus V To Bus V  
37     500.00   230.00  
38     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
39  Solar = 100 MW, Hydro = 500 MW, LLS       202  EAST500     500.00     203   
40  Solar = 100 MW, Hydro = 500 MW, LLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
39  EAST230     230.00       1              203        765          642.25   
40  SUB230      230.00       1              205        810          687.22   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
39            218.43      678.38     1.76      71.35  800.0  82.83  84.80   
40            131.29      699.65     1.48      73.96  800.0  87.77  87.46   

   From Bus V To Bus V  
39     500.00   230.00  
40     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
41  Solar = 100 MW, Hydro = 500 MW, RLS       202  EAST500     500.00     203   
42  Solar = 100 MW, Hydro = 500 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
41  EAST230     230.00       1              203        803          660.86   
42  SUB230      230.00       1              205        845          702.86   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
41            246.29      705.27     1.94      78.70  800.0  86.99  88.16   
42            160.55      720.96     1.61      80.44  800.0  91.54  90.12   

   From Bus V To Bus V  
41     500.00   230.00  
42     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
43  Solar = 100 MW, Hydro = 500 MW, HLS       202  EAST500     500.00     203   
44  Solar = 100 MW, Hydro = 500 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
43  EAST230     230.00       1              203        851          675.54   
44  SUB230      230.00       1              205        891          714.78   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
43            291.71      735.84     2.17      88.33  800.0  92.16  91.98   
44            210.64      745.17     1.79      89.42  800.0  96.51  93.15   

   From Bus V To Bus V  
43     500.00   230.00  
44     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
45  Solar = 100 MW, Hydro = 600 MW, LLS       202  EAST500     500.00     203   
46  Solar = 100 MW, Hydro = 600 MW, LLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
45  EAST230     230.00       1              203        828          697.49   
46  SUB230      230.00       1              205        901          763.85   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
45            220.12      731.40     2.06      83.68  800.0  89.70  91.42   
46            121.77      773.49     1.83      91.35  800.0  97.55  96.69   

   From Bus V To Bus V  
45     500.00   230.00  
46     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
47  Solar = 100 MW, Hydro = 600 MW, RLS       202  EAST500     500.00     203   
48  Solar = 100 MW, Hydro = 600 MW, RLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
47  EAST230     230.00       1              203        882          715.23   
48  SUB230      230.00       1              205        952          778.53   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I   %MVA  \
47            273.07      765.59     2.34      95.00  800.0   95.57  95.70   
48            179.91      799.04     2.04     102.08  800.0  103.12  99.88   

   From Bus V To Bus V  
47     500.00   230.00  
48     500.00   230.00

Scenario  From Bus        From Bus Ext  To Bus  \
49  Solar = 100 MW, Hydro = 600 MW, HLS       202  EAST500     500.00     203   
50  Solar = 100 MW, Hydro = 600 MW, HLS       204  SUB500      500.00     205   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
49  EAST230     230.00       1              203        934          729.65   
50  SUB230      230.00       1              205       1003          790.15   

    Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate      %I    %MVA  \
49            320.44      796.91     2.62     106.42  800.0  101.16   99.61   
50            231.20      823.28     2.26     113.19  800.0  108.58  102.91   

   From Bus V To Bus V  
49     500.00   230.00  
50     500.00   230.00

##### Checking branch loading greater than 80% 

In [26]:
branch_loading = data_all[data_all['From Bus V'] == data_all['To Bus V']].reset_index(drop=True)
branch_overload = branch_loading[branch_loading['%I'] > 80.0].reset_index(drop=True)
branch_overload['Branch'] = branch_overload['From Bus'].astype(str) + ' ' + branch_overload['From Bus Ext'] + ' to ' + \
branch_overload['To Bus'].astype(str) + ' '+ branch_overload['To Bus Ext'] + ' ' + branch_overload['Ckt ID'].astype(str)
list_branch = branch_overload['Branch'].unique()
len(list_branch)
print(list_branch)
# for branch in list_branch:
#     load = list(branch_overload[branch_overload['Branch'] == branch])
#     display(len(load))


['153 MID230      230.00 to 154 DOWNTN      230.00 1'
 '3001 MINE        230.00 to 3003 S. MINE     230.00 1'
 '3005 WEST        230.00 to 3007 RURAL       230.00 1'
 '153 MID230      230.00 to 154 DOWNTN      230.00 2'
 '154 DOWNTN      230.00 to 203 EAST230     230.00 1']


##### Voltage Upper Limits 

In [6]:
list_volt = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
list_gen_hydro = [400,500,600] #dry, average, wet hydological scenarios
for gen in list_gens:
    for gen_hy in list_gen_hydro:
        for lsc in list_lsc:
            file_sav_in = 'savnw_sol_' + str(gen) +'_hy_' +str(gen_hy) +'_' + str(lsc)+'_volt.csv'
        data_volt = pd.read_csv(file_sav_in,header=1, skiprows=0)
        data_volt['Scenario']= 'Solar = ' + str(gen) + ' MW, ' +'Hydro = ' +str(gen_hy) +' MW, ' + lsc.upper()
        list_volt.append(data_volt)
data_all_volt = pd.concat(list_volt).reset_index(drop=True)
data_all_volt = data_all_volt[['Scenario','Bus Number', 'Bus Name', 'Area', 'Base Voltage', 'Bus Voltage(PU)','Bus Voltage(kV)']]

In [7]:
data_all_volt[data_all_volt['Bus Voltage(PU)']>1.05].reset_index(drop=True)

Scenario  Bus Number      Bus Name  Area  \
0     Solar = 0 MW, Hydro = 500 MW, HLS         211  HYDRO_G          2   
1     Solar = 0 MW, Hydro = 500 MW, HLS         212  HYDRO_N          2   
2     Solar = 0 MW, Hydro = 600 MW, HLS         211  HYDRO_G          2   
3     Solar = 0 MW, Hydro = 600 MW, HLS         212  HYDRO_N          2   
4    Solar = 50 MW, Hydro = 500 MW, HLS         211  HYDRO_G          2   
5    Solar = 50 MW, Hydro = 500 MW, HLS         212  HYDRO_N          2   
6    Solar = 50 MW, Hydro = 600 MW, HLS         211  HYDRO_G          2   
7    Solar = 50 MW, Hydro = 600 MW, HLS         212  HYDRO_N          2   
8   Solar = 100 MW, Hydro = 500 MW, HLS         211  HYDRO_G          2   
9   Solar = 100 MW, Hydro = 500 MW, HLS         212  HYDRO_N          2   
10  Solar = 100 MW, Hydro = 600 MW, HLS         211  HYDRO_G          2   
11  Solar = 100 MW, Hydro = 600 MW, HLS         212  HYDRO_N          2   

    Base Voltage  Bus Voltage(PU)  Bus Voltage(kV)  
0           20.0            1.061           21.228  
1           20.0            1.061           21.228  
2           20.0            1.080           21.609  
3           20.0            1.080           21.609  
4           20.0            1.057           21.131  
5           20.0            1.057           21.131  
6           20.0            1.075           21.495  
7           20.0            1.075           21.495  
8           20.0            1.053           21.053  
9           20.0            1.053           21.053  
10          20.0            1.070           21.403  
11          20.0            1.070           21.403

##### Voltage Lower Limits

In [8]:
data_all_volt[data_all_volt['Bus Voltage(PU)']<0.95]

Scenario  Bus Number      Bus Name  Area  \
6      Solar = 0 MW, Hydro = 400 MW, HLS         154  DOWNTN           1   
11     Solar = 0 MW, Hydro = 400 MW, HLS         205  SUB230           2   
27     Solar = 0 MW, Hydro = 500 MW, HLS         103  SOLAR_PV         1   
31     Solar = 0 MW, Hydro = 500 MW, HLS         154  DOWNTN           1   
34     Solar = 0 MW, Hydro = 500 MW, HLS         203  EAST230          2   
36     Solar = 0 MW, Hydro = 500 MW, HLS         205  SUB230           2   
46     Solar = 0 MW, Hydro = 500 MW, HLS        3007  RURAL            5   
47     Solar = 0 MW, Hydro = 500 MW, HLS        3008  CATDOG           5   
52     Solar = 0 MW, Hydro = 600 MW, HLS         103  SOLAR_PV         1   
56     Solar = 0 MW, Hydro = 600 MW, HLS         154  DOWNTN           1   
59     Solar = 0 MW, Hydro = 600 MW, HLS         203  EAST230          2   
60     Solar = 0 MW, Hydro = 600 MW, HLS         204  SUB500           2   
61     Solar = 0 MW, Hydro = 600 MW, HLS         205  SUB230           2   
71     Solar = 0 MW, Hydro = 600 MW, HLS        3007  RURAL            5   
72     Solar = 0 MW, Hydro = 600 MW, HLS        3008  CATDOG           5   
81    Solar = 50 MW, Hydro = 400 MW, HLS         154  DOWNTN           1   
86    Solar = 50 MW, Hydro = 400 MW, HLS         205  SUB230           2   
106   Solar = 50 MW, Hydro = 500 MW, HLS         154  DOWNTN           1   
111   Solar = 50 MW, Hydro = 500 MW, HLS         205  SUB230           2   
121   Solar = 50 MW, Hydro = 500 MW, HLS        3007  RURAL            5   
122   Solar = 50 MW, Hydro = 500 MW, HLS        3008  CATDOG           5   
127   Solar = 50 MW, Hydro = 600 MW, HLS         103  SOLAR_PV         1   
131   Solar = 50 MW, Hydro = 600 MW, HLS         154  DOWNTN           1   
134   Solar = 50 MW, Hydro = 600 MW, HLS         203  EAST230          2   
135   Solar = 50 MW, Hydro = 600 MW, HLS         204  SUB500           2   
136   Solar = 50 MW, Hydro = 600 MW, HLS         205  SUB230           2   
146   Solar = 50 MW, Hydro = 600 MW, HLS        3007  RURAL            5   
147   Solar = 50 MW, Hydro = 600 MW, HLS        3008  CATDOG           5   
156  Solar = 100 MW, Hydro = 400 MW, HLS         154  DOWNTN           1   
181  Solar = 100 MW, Hydro = 500 MW, HLS         154  DOWNTN           1   
186  Solar = 100 MW, Hydro = 500 MW, HLS         205  SUB230           2   
197  Solar = 100 MW, Hydro = 500 MW, HLS        3008  CATDOG           5   
202  Solar = 100 MW, Hydro = 600 MW, HLS         103  SOLAR_PV         1   
206  Solar = 100 MW, Hydro = 600 MW, HLS         154  DOWNTN           1   
209  Solar = 100 MW, Hydro = 600 MW, HLS         203  EAST230          2   
210  Solar = 100 MW, Hydro = 600 MW, HLS         204  SUB500           2   
211  Solar = 100 MW, Hydro = 600 MW, HLS         205  SUB230           2   
221  Solar = 100 MW, Hydro = 600 MW, HLS        3007  RURAL            5   
222  Solar = 100 MW, Hydro = 600 MW, HLS        3008  CATDOG           5   

     Base Voltage  Bus Voltage(PU)  Bus Voltage(kV)  
6           230.0            0.931          214.213  
11          230.0            0.941          216.472  
27           13.8            0.946           13.057  
31          230.0            0.916          210.790  
34          230.0            0.945          217.303  
36          230.0            0.926          213.079  
46          230.0            0.945          217.277  
47          230.0            0.938          215.770  
52           13.8            0.925           12.766  
56          230.0            0.895          205.796  
59          230.0            0.925          212.816  
60          500.0            0.938          468.760  
61          230.0            0.905          208.088  
71          230.0            0.928          213.507  
72          230.0            0.920          211.526  
81          230.0            0.937          215.425  
86          230.0            0.946          217.659  
106         230.0  

#### Conclusion 

Year 1 Toplogy limit violation check for overloading and voltage was carried out for the three load scenarios. It was seen that
1. overloading of greater than 80% was reported for following branches and transformers for considered solar output and load scenarios 
    * Solar = 0 MW, HLS \
      Branch 153 MID230 230.00 154 DOWNTN 230.00 Ckt ID 1
    * Solar = 0 MW, HLS	\
    Transformer 205	SUB230 230.00	206	URBGEN 18.000
    * Solar = 50 MW, HLS	\
    Transformer 205	SUB230 230.00	206	URBGEN 18.000
    * Solar = 100 MW, HLS	\
    Transformer 205	SUB230 230.00	206	URBGEN 18.000
2. The buses violating the upper voltage limit for various scenarios are:
    * Solar = 0 MW, Scenarios = LLS, RLS \
      BUS 201, 308
    * Solar = 0 MW, Scenarios = HLS

      BUS 201
    * Solar = 50 MW, Scenarios = LLS, RLS \
      BUS 201, 308
    * Solar = 50 MW, Scenarios = HLS \
      BUS 201
    * Solar = 100 MW, Scenarios = LLS, RLS \
      BUS 201, 308
    * Solar = 100 MW, Scenarios = HLS 
    
      BUS 201